File: Algorithms/Strings/Gemstones.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'gemstones' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts STRING_ARRAY arr as parameter.
     */

    public static int gemstones(List<string> arr)
    {
        HashSet<char> gemstonesSet = new HashSet<char>(arr[0]);

        for (int i = 1; i < arr.Count; i++)
        {
            gemstonesSet.IntersectWith(arr[i]);
        }

        return gemstonesSet.Count;
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<string> arr = new List<string>();

        for (int i = 0; i < n; i++)
        {
            string arrItem = Console.ReadLine();
            arr.Add(arrItem);
        }

        int result = Result.gemstones(arr);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Algorithms/Strings/Gemstones.php
<?php

/*
 * Complete the 'gemstones' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts STRING_ARRAY arr as parameter.
 */

function gemstones($arr) {
    # Write your code here
    $common_elements = array_fill_keys(range('a', 'z'), True);

    foreach ($arr as $rock) {
        $current_rock_set = array_fill_keys(str_split($rock), True);

        foreach ($common_elements as $char => $val) {
            if (!isset($current_rock_set[$char])) {
                unset($common_elements[$char]);
            }
        }
    }
    return count($common_elements);
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));

$arr = array();

for ($i = 0; $i < $n; $i++) {
    $arr_item = rtrim(fgets(STDIN), "\r\n");
    $arr[] = $arr_item;
}

$result = gemstones($arr);

fwrite($fptr, $result . "\n");
fclose($fptr);
